key.core/src/test/java/de/uka/ilkd/key/speclang/njml/JMLParserExceptionTest.java

```
@@ -0,0 +1,153 @@
+        assert fileURL != null : "Directory 'exceptional' not found";
+        assert fileURL.getProtocol().equals("file") : "Test resources must be in file system";
+        Path dir = Paths.get(fileURL.toURI());
+        return Files.walk(dir).filter(it -> it.getFileName().toString().endsWith(".java"))
+                .map(it -> Arguments.of(it, it.getFileName()));
+    }
+
+
+    @ParameterizedTest(name = "case {1}")
+    @MethodSource("getFiles")
+    public void testParseAndInterpret(Path file, Path localFilename) throws Exception {
+        parseAndInterpret(file);
+    }
+
+    // This method does not depend on anything can also be called from other test cases.
+    public static void parseAndInterpret(Path file) throws Exception {
+
+        List<String> lines = Files.readAllLines(file);
+        Properties props = new Properties();
+        for (String line : lines) {
+            Matcher m = PROP_LINE.matcher(line);
+            if (m.matches()) {
+                props.put(m.group(1), m.group(2));
+            } else {
+                break;
+            }
+        }
+
+        if ("true".equals(props.get("ignore"))
+                || IGNORE_BROKEN && "true".equals(props.get("broken"))) {
+            Assumptions.abort("This test case has been marked to be ignored");
+        }
+
+        try {
+            ProblemLoaderControl control = new DefaultUserInterfaceControl(null);
+            AbstractProblemLoader pl = new SingleThreadProblemLoader(file.toFile(), null, null,
+                null, AbstractProfile.getDefaultProfile(), false,
+                control, false, new Properties());
+            pl.setLoadSingleJavaFile(true);
+            pl.load();
+            // No exception encountered
+            assertEquals("true", props.get("noException"),
+                "Unless 'noException: true' has been set, an exception is expected");
+
+        } catch (Throwable e) {
```

Maybe you should not catch `Throwable`. The logic gets quite complicated. 

JUnit 5 provides `assertThrows` to specify that a code block should throw an exception. https://howtodoinjava.com/junit5/expected-exception-example/

This only misses a handler for the exceptional case. The source code is [here](https://github.com/junit-team/junit5/blob/main/junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertThrows.java#L49) and could be extended. 

You could also exclude the assertion case from `Throwable` by adding:

```java
try{...}
catch(AssertionFailure e){throw e;}
catch(Throwable e){ ... }
```

---

right!